### DISCUSSION TITLE
How to set a max character limit?

### DISCUSSION THREAD

**Discussion by faizanje** [singerdmx/flutter-quill#860] [Mobile] [answered]
I need to set a max character limit of let's say 1000 characters. I can get the length of the character using **quillController.document.length** and then added a listener which checks the limit but I can't figure out how to restrict the addition of new characters

**Comment by Arenukvern** [comment, marked as answer]
I managed to solve it by attaching listener and changing the document the following way:

in initState for example:
```dart
widget.quillController.document.changes.listen(_onDocumentChange)
```

somewhere in class:
```dart
void _onDocumentChange(Tuple3<quill.Delta, quill.Delta, quill.ChangeSource> tuple) {
    final documentLength = widget.quillController.document.length;
    if (documentLength > widget.limit) {
      final latestIndex = widget.limit - 1;
      widget.quillController.replaceText(
        latestIndex,
        documentLength - widget.limit,
        '',
        TextSelection.collapsed(offset: latestIndex),
      );
    }
 }
```

**Reply by singerdmx** [reply]
Or you can submit a PR

**Comment by TDA-TRAN-VIET** [comment]
if you typing icon, there will be problems with logic
it's bug: Invalid argument(s): string is not well-formed UTF-16
